### PR TITLE
Returns name for given column in the row.

### PR DESCRIPTION
**Pull request by ionelanton**
Returns name for given column in the row when using db_postgres.

### CODE REVIEW

File: lib/impure/db_postgres.nim [hunk 1]
```diff
@@ -234,6 +234,10 @@ proc len*(row: InstantRow): int32 {.inline.} =
   ## returns number of columns in the row
   pqNfields(row.res)
 
+proc getColumnName*(row: InstantRow, col: int32) : string {.inline.} =
```
Comment (Araq) [inline lib/impure/db_postgres.nim]: I think this should take a 'col: int' or 'col: Natural'.

Comment (ionelanton) [inline lib/impure/db_postgres.nim]: In this case, the signature of 
proc pqfname*(res: PPGresult, field_num: int32)
in postgres.nim must change to
proc pqfname*(res: PPGresult, field_num: int)

Comment (Araq) [inline lib/impure/db_postgres.nim]: nah, you convert it to int32 internally, in the proc. You cannot just change the interface that Postgres exposes.

Comment (ionelanton) [inline lib/impure/db_postgres.nim]: Done.

**Comment by Araq** [comment]
The interface that db_postgres should expose is actually:

```nim

iterator instantRows*(db: DbConn; columns: var DbColumns; query: SqlQuery;
                      args: varargs[string, `$`]): InstantRow = ...

```

If one of the db_ modules has ``getColumnName`` that the others lack, that's pretty bad.

**Comment by ionelanton** [comment]
I will look into it latter today.

**Comment by ionelanton** [comment]
I implemented the interface but I have the following problem:

When applying `[]` operator to an InstantRow object, as defined in db_postgres.nim I should get a string value, the value of the cell stored in the DB.
```
proc `[]`*(row: InstantRow; col: int32): string {.inline.} =
  ## returns text for given column of the row
  $pqgetvalue(row.res, row.line, col)
```
Instead, the returned type is PPGresult.
```
for row in db.instantRows(sql"SELECT * FROM test_table"):
      echo row[0].type.name #PPGresult 
```
When I use the proc I wrote, getColumnValue, I get the value stored in DB
```
for row in db.instantRows(cols, sql"SELECT * FROM test_table"):
      echo row.getColumnValue(3) # stored value in the DB
```
I think there is a problem (bug) with `[]` operator overloading when applied to an InstantRow object.
I can't proceed with the pull request because of this problem.

I'm using Nim Compiler Version 0.16.1 <2017-02-11> [Windows: i386]

@Araq I'm waiting for your feedback.
Thank you!

**Comment by Araq** [comment]
> Instead, the returned type is PPGresult.

That is very strange, push the code to some branch and let me have a look.

> I think there is a problem (bug) with [] operator overloading when applied to an InstantRow object.

I really doubt it, overloading of '[]' is used everywhere in Nim.

**Comment by Araq** [comment]
Sorry, just read the code. ``InstantRow``'s definition is wrong for this to work. Tuples have a builtin [] operator that is preferred. Change its definition to ``object`` please.

**Comment by ionelanton** [comment]
I'll make a new pull request.

**Pull request closed without merging**